.github/workflows/nightly.yml

```diff
@@ -0,0 +1,63 @@
+name: Nightly Build
+on:
+  schedule:
+    - cron: '0 1 * * *'
+  workflow_dispatch:
+
+jobs:
+  should-build-change:
+    runs-on: ubuntu-latest
+    outputs:
+      repo-cache-hit: ${{ steps.cache-last-commit.outputs.cache-hit }}
+    steps:
+      - name: Fetch Sources
+        uses: actions/checkout@v4
+      - run: |
+          git rev-parse HEAD >> lastCommit
+      - name: Check New Changes
+        id: cache-last-commit
+        uses: actions/cache@v4
+        with:
+          path: lastCommit
+          key: lastCommit-${{ hashFiles('lastCommit') }}
+
+  # Prepare and publish the plugin to JetBrains Marketplace repository
+  pre-release:
+    needs: should-build-change
+    if: ${{ needs.should-build-change.outputs.repo-cache-hit != 'true' || github.event_name != 'schedule' }}
+    name: Publish Plugin
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      # Check out current repository
+      - name: Fetch Sources
+        uses: actions/checkout@v4
+
+      # Set up Java environment for the next steps
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'temurin'
+          cache: 'gradle'
+
+      # Setup Gradle
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@af1da67850ed9a4cedd57bfd976089dd991e2582 # v4.0.0
```

Comment:
why use a sha instead of a version?

Reply:
you should always use sha1 versions instead of tags, for non-trusted orgs.
A tag can change. If a malicious actor publishes a vulnerable action under a known/existing tag. your repository would be at risk. A commit sha is immutable, thus more secure.